lib/concurrent/synchronization/immutable_struct.rb

```diff
@@ -0,0 +1,67 @@
+module Concurrent
+  module Synchronization
+    # Similar to Struct but the fields are immutable and always visible (like Java final fields)
+    # @example
+    #   Person = ImmutableStruct.with_fields :name, :age
+    #   Person.new 'John Doe', 15
+    #   Person['John Doe', 15]
+    #   Person['John Doe', 15].members # => [:name, :age]
+    #   Person['John Doe', 15].values  # => ['John Doe', 15]
+    class ImmutableStruct < Synchronization::Object
+      def self.with_fields(*names, &block)
```

Comment:
I'll work on it tonight. I'll be in a hotel room looking for something productive to do (I'm attending [Stir Trek](http://stirtrek.com/) tomorrow).